### ISSUE TITLE
Bug - Menus (dropdown, menutoggle, etc.) - first menu item in a dropdown has a highlight by default (and it shouldn't)

### ISSUE THREAD

**Issue by garrett** [patternfly/patternfly#6571]
**Describe the problem**
This is an issue with 
- [x] Patternfly 5  

[?] Patternfly 6  

Please give a clear and concise description of the problem. Which components are affected?

When clicking on a dropdown or menutoggle, the first item in a menu is already selected.

**How do you reproduce the problem?**

Open any dropdown, menutoggle, or other menu-based component and look at the first menu item. It looks like it's being hovered, but it isn't. This affects all the new menus on the PatternFly website as well.

**Expected behavior**

An item that isn't hovered or focused shouldn't look hovered or focused.

**Is this issue blocking you?**

We had this as part of a blocker for our release, as it's a regression from the previous (now deprecated) dropdowns and we recently ported to the new ones.

**Screenshots**

From PatternFly's website:

![image](https://github.com/patternfly/patternfly/assets/10246/e569009d-88ec-4d34-a52c-48a312063f4d)

![image](https://github.com/patternfly/patternfly/assets/10246/4b76e2b8-1dba-484a-ab77-708ab3259678)

![image](https://github.com/patternfly/patternfly/assets/10246/e754d886-1063-40db-88b6-aae253c3143a)

![image](https://github.com/patternfly/patternfly/assets/10246/2300d65e-160e-4f04-9904-0d4c5b1f6193)

![image](https://github.com/patternfly/patternfly/assets/10246/be56ea3d-4342-4770-9daa-f502e329ab31)

This affects Chrome too:

![image](https://github.com/patternfly/patternfly/assets/10246/2b5c79bb-7d9c-4bb9-9109-6ed2d058baf7)

And also WebKit:

![image](https://github.com/patternfly/patternfly/assets/10246/0a9405cf-9ec4-4a35-a50a-2b03c23e01b0)

**What is your environment?**
 - OS: Fedora Linux 40
 - Browser: Firefox 125.0.1, Chromium 124.0.6367.60, GNOME Web 46 (WebKitGTK 2.44.1, basically Safari 17 on Linux)

**What is your product and what release date are you targeting?**

Cockpit. Yesterday was the release. We're unfortunately now shipping with this bug, as we cannot indefinitely postpone the release to wait on a fix in a PatternFly release. :frowning_face:

**Comment by thatblindgeye** [comment]
@garrett thanks for bringing this to our attention!

For what it's worth, the first item upon clicking open a toggle is being focused rather than being styled to look hovered. The focus and hover styles are the same, though, which may be adding to this confusion. This functionality was added in React in a PR to [focus the first item when opened by click](https://github.com/patternfly/patternfly-react/pull/10125), which helped resolve an issue regarding [not being able to traverse the menu items](https://github.com/patternfly/patternfly-react/issues/9788) via keyboard after clicking the toggle.

With that in mind does that help alleviate the concern with this behavior, or is the behavior still unexpected and/or confusing?

**Comment by martnaum** [comment]
This also seems to cause a problem with fields that allow text input via the keyboard. Previously, we were able to click directly into the field and filter the displayed elements via keyboard input. This is now only possible after an additional click in the field. You can easily see that by comparing the UX of https://www.patternfly.org/components/menus/select/#typeahead to:

* https://www.patternfly.org/2022.05/components/select/#typeahead
or
* https://staging-v6.patternfly.org/components/menus/select#typeahead
